tool/tsh/resolve_default_addr_test.go

```diff
@@ -259,3 +259,12 @@ func TestResolveDefaultAddrTimeoutBeforeAllRacersLaunched(t *testing.T) {
 	// the call timing out.
 	require.Equal(t, context.DeadlineExceeded, err)
 }
+
+func TestResolveDefaultAddrHTTPSProxy(t *testing.T) {
+	// Given an http proxy address...
+	t.Setenv("HTTPS_PROXY", "http://fakeproxy.example.com")
+	// When I attempt to resove an address...
+	_, err := pickDefaultAddr(context.Background(), true, "example.com", []int{9999})
+	// Expect that pickDefaultAddr uses the http proxy.
+	require.ErrorContains(t, err, "lookup fakeproxy.example.com: no such host")
```

Comment:
Hmm, this seems a bit fragile as it's actually going to try to connect to fakeproxy.example.com and depends on getting a DNS error.

If example.com decides to register a wildcard DNS record, our test breaks.

Reply:
`example.com` is a reserved name so I don't think it's likely to ever change, though there are some whole TLDs that are reserved for testing purposes: https://www.rfc-editor.org/rfc/rfc2606.html#section-2

Reply:
Sure, but I guess what I'm getting at is that we're not actually testing the behavior that we want (that `pickDefaultAddr` successfully dials through a proxy), we're kind of testing a side-effect of that instead.